Bottom: 9165d26e73d8829a410bf6a1b2266c004cfeebfb
Top:    979c1d3313190ed7ef8d5862e5364385b34ab4b7
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-21 11:51:13 -0500

Rework check for first install

Create a fact for nexus_first_install rather than checking stdout
of a command in several places.



---
diff --git a/roles/nexus3/tasks/main.yml b/roles/nexus3/tasks/main.yml
index 1cfa71113..0e570bf4c 100644
--- a/roles/nexus3/tasks/main.yml
+++ b/roles/nexus3/tasks/main.yml
@@ -32,7 +32,7 @@
     - nuget-group
     - nuget-hosted
     - nuget.org-proxy
-  when: nexus_data_dir_contents.stdout == "" and nexus_delete_default_repos
+  when: nexus_first_install and nexus_delete_default_repos
 
 - name: Deleting default blobstore
   include: delete_blobstore_each.yml
@@ -48,7 +48,7 @@
     - name: "{{ nexus_blob_names.mvn.blob }}"
     - name: "{{ nexus_blob_names.gitlfs.blob }}"
     - name: "{{ nexus_blob_names.yum.blob }}"
-  when: nexus_data_dir_contents.stdout == "" and nexus_delete_default_blobstore
+  when: nexus_first_install and nexus_delete_default_blobstore
 
 - include: setup_ldap_each.yml
   with_items: "{{ ldap_connections }}"
diff --git a/roles/nexus3/tasks/nexus_install.yml b/roles/nexus3/tasks/nexus_install.yml
index b3bf57b59..02053005c 100644
--- a/roles/nexus3/tasks/nexus_install.yml
+++ b/roles/nexus3/tasks/nexus_install.yml
@@ -210,11 +210,14 @@
   check_mode: no
   changed_when: false
 
+- set_fact:
+    nexus_first_install: "{{ nexus_data_dir_contents.stdout == '' }}"
+
 - name: Clean cache for upgrade process
   file:
     path: "{{ nexus_data_dir }}/clean_cache"
     state: touch
-  when: nexus_latest_version.changed and nexus_data_dir_contents.stdout != ""
+  when: nexus_latest_version.changed and not nexus_first_install
   tags:
     # hard to run as a handler for time being
     - skip_ansible_lint
@@ -257,12 +260,12 @@
 - name: First-time install admin password
   set_fact:
     current_nexus_admin_password: 'admin123'
-  when: nexus_data_dir_contents.stdout == ""
+  when: nexus_first_install
 
 - name: Subsequent re-provision admin password
   set_fact:
     current_nexus_admin_password: "{{ nexus_admin_password }}"
-  when: nexus_data_dir_contents.stdout != ""
+  when: not nexus_first_install
   no_log: true
 
 - name: Create directory to hold current groovy scripts for reference
